## AIN212 Project
#### Talha Kaba - 2210765037
#### Abdulkadir Parlak - 2210765025
### Dataset: Fatalities in the Israeli-Palestinian
* source: https://www.kaggle.com/datasets/willianoliveiragibin/fatalities-in-the-israeli-palestinian

In [15]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [22]:
df = pd.read_csv('fatalities_isr_pse_conflict_2000_to_2023.csv')
df.head()

age_range = df['age'].min(), df['age'].max()
print("Age Range:", age_range)

categorical_columns = df.select_dtypes(include=['object']).columns
for column in categorical_columns:
    if(column == 'notes'):
        continue
    categories = df[column].unique()
    print("Categories in", column, ":", categories)


distinct_locations = df['event_location'].nunique()
print("Number of distinct event locations:", distinct_locations)

categorical_columns = df.select_dtypes(include=['object']).columns
for column in categorical_columns:

    distinct_variables = df[column].nunique()
    print("Number of distinct variables in", column, ":", distinct_variables)


filtered_df = df[df['age'] <= 18]
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)





Age Range: (1.0, 112.0)
Categories in name : ["'Abd a-Rahman Suleiman Muhammad Abu Daghash"
 "Usayed Farhan Muhammad 'Ali Abu 'Ali" "'Abdallah 'Imad Sa'ed Abu Hassan"
 ... 'Hillel Lieberman' "Fahed Mustafa 'Odeh Baker" 'Wichlav Zalsevsky']
Categories in date_of_event : ['2023-09-24' '2023-09-22' '2023-09-20' ... '2000-10-17' '2000-10-07'
 '2000-10-02']
Categories in citizenship : ['Palestinian' 'Israeli' 'Jordanian' 'American']
Categories in event_location : ['Nur Shams R.C.' 'Kfar Dan' "'Aqbat Jaber R.C." 'Jenin R.C.' 'Gaza City'
 "al-'Arrub R.C." 'Argaman' "'Aqqabah" 'Beit Sira' 'al-Musrarah'
 'a-Zababidah' 'Beit Hagai' 'Huwarah' 'Jenin' 'Balata R.C.' 'Tulkarm R.C.'
 'Nablus' 'Bir al-Basha' 'Tel Aviv-Yafo' 'Burqah' 'Tulkarm' 'Ofra'
 "Ma'ale Adumim" 'Mitzpe Eshtamoa' "'Ein Beit al-Maa R.C." 'Sabastiya'
 'Um Safa' 'Deir Nizam' 'al-Birah' 'Qalandia' "'Askar R.C." 'Turmusaya'
 'Eli' 'Tubas' 'Husan' 'Rantis' 'a-Nabi Saleh' 'Hermesh' 'Tene'
 "Barta'ah a-Sharqiyah" 'Khan Yunis' 'Rehovot' 'B

## Information About Dataset
### Labels
* name: 11083 distinct names.
*  took_part_in_the_hostilities : [nan 'No' 'Yes' 'Unknown' 'Israelis' 'Object of targeted killing']
*  date_of_event: YYYY-MM-DD,  # of distinct dates 2405
*  date_of_death: YYYY-MM-DD,  # of distinct dates 2593
*  age: [1:112]
* event_location_region : ['West Bank' 'Gaza Strip' 'Israel']
* citizenship: ['Palestinian' 'Israeli' 'Jordanian' 'American']
* event_location: number of distinct locations = 494
* killed_by : ['Israeli security forces' 'Palestinian civilians' 'Israeli civilians']
* place_of_residence_district : 20 distinct place_districts
* place_of_residence : 580 distinct places.
* type_of_injury : 13 distinct types of injury.
* ammunition : 21 distinct ammunations.
* killed_by : ['Israeli security forces' 'Palestinian civilians' 'Israeli civilians']
* notes: 6744 distinct notes.